File: python/2411.smallest-subarrays-with-maximum-bitwise-or.py
class Solution:
    def smallestSubarrays(self, nums: list[int]) -> list[int]:
        n = len(nums)
        ans = [1] * n

        for i in range(n):
            j = i - 1
            v = nums[i]

            while j >= 0 and (v | nums[j]) != nums[j]:
                ans[j] = i - j + 1
                nums[j] |= v
                j -= 1

        return ans

File: cpp/2411.smallest-subarrays-with-maximum-bitwise-or.cpp
#include <vector>
using std::vector;

class Solution {
public:
  vector<int> smallestSubarrays(vector<int> &nums) {
    const int n = nums.size();
    vector<int> ans(n, 1);

    for (int i = 0; i < n; i++) {
      int x = nums[i];
      int j = i - 1;

      while (j >= 0 && (x | nums[j]) != nums[j]) {
        nums[j] |= x;
        ans[j] = i - j + 1;
        j--;
      }
    }

    return ans;
  }
};